code/datums/brain_damage/imaginary_friend.dm

```diff
@@ -87,12 +85,21 @@
 		client.images |= current_image
 
 /mob/abstract/mental/friend/Destroy()
+	SSghostroles.remove_spawn_atom("friend", src)
 	if(owner.client)
 		owner.client.images.Remove(human_image)
 	if(client)
 		client.images.Remove(human_image)
 	return ..()
 
+/mob/abstract/mental/friend/proc/spawn_into_friend(var/mob/user)
+	if(owner.stat == DEAD)
+		to_chat(user, SPAN_WARNING("\The [owner] is dead! You cannot be their friend now."))
+		return
+	ckey = user.ckey
+	qdel(user)
```

Comment:
Afaik most ghostspanwers do not delete user after the assignment atm.
(And if that is required it should be handled in a centralized location instead of locally per spawn proc)